modules/solid_mechanics/src/materials/RadialReturnStressUpdate.C

```diff
@@ -250,21 +261,32 @@ RadialReturnStressUpdateTempl<is_ad>::updateState(
   // configuration
   GenericRankTwoTensor<is_ad> deviatoric_trial_stress = stress_new.deviatoric();
 
-  // compute the effective trial stress
+  // compute the effective trial stress.  Ternary because the sqrt is not differentiable at 0
   GenericReal<is_ad> dev_trial_stress_squared =
       deviatoric_trial_stress.doubleContraction(deviatoric_trial_stress);
   GenericReal<is_ad> effective_trial_stress = MetaPhysicL::raw_value(dev_trial_stress_squared)
                                                   ? std::sqrt(3.0 / 2.0 * dev_trial_stress_squared)
                                                   : 0.0;
+  // Initialize models around a stress that uses the previous inelastic strain increment to
+  // predict the final stress state for this step. (Dunne & Petrinic eq 5.16 )
+  // fixme three shear mod is set in comptueStressInitialize
+  Real predicted_eff_strain_inc =
+      _scale_strain_predictor * _eff_inelastic_strain_rate_old[_qp] * _dt;
+
+  const Real three_shear_modulus =
+      3.0 *
+      MetaPhysicL::raw_value(ElasticityTensorTools::getIsotropicShearModulus(elasticity_tensor));
+  const GenericReal<is_ad> predictor_effective_trial_stress =
+      effective_trial_stress - three_shear_modulus * predicted_eff_strain_inc;
 
-  computeStressInitialize(effective_trial_stress, elasticity_tensor);
+  computeStressInitialize(predictor_effective_trial_stress, elasticity_tensor);
 
   mooseAssert(
       _three_shear_modulus != 0.0,
       "Shear modulus is zero. Ensure that the base class computeStressInitialize() is called.");
 
   // Use Newton iteration to determine the scalar effective inelastic strain increment
-  _effective_inelastic_strain_increment = 0.0;
+  _effective_inelastic_strain_increment = predicted_eff_strain_inc;
```

Comment:
```suggestion
  _effective_inelastic_strain_increment = predicted_eff_strain_increment;
```

everywhere

Reply:
I thought we could use abbreviations for local variables.